src/utc_offset.rs

```diff
@@ -448,8 +452,28 @@ fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
 
         let diff_secs = filetime_to_secs(&ft_local) - filetime_to_secs(&ft_system);
 
-        diff_secs.try_into().ok().map(UtcOffset::seconds)
+        return diff_secs.try_into().ok().map(UtcOffset::seconds);
     }
+
+    #[cfg(target_arch = "wasm32")]
+    {
+        let timestamp_utc: i64 = datetime.timestamp();
+        let t1 = (timestamp_utc & 0xFFFFFFFF) as i32;
+        let t2 = (timestamp_utc >> 32) as i32;
+        let timezone_offset = js! {
+            var ts_utc = @{t2} << 32 + @{t1};
+            var d = new Date(ts_utc);
+            return d.getTimezoneOffset() * -60;
+        };
+        if let Ok(seconds) = stdweb::unstable::TryInto::try_into(timezone_offset) {
+            return Some(UtcOffset::seconds(seconds));
+        } else {
+            return None;
+        }
+    }
+
+    #[allow(unreachable_code)]
+    None
```

Comment:
I'd prefer to avoid having this just floating around, especially with allowing that lint. I think it would be cleaner to aggressively gate it.

`cfg_if` will likely be helpful here, especially as the number of targets increases.